### PR TITLE
Add proper xmldoc for RetryAttribute

### PR DESCRIPTION
**Pull request by MhdTlb**
Correct xmldoc to be similar to the documentation

### CODE REVIEW

File: src/NUnitFramework/framework/Attributes/RetryAttribute.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@
 // ***********************************************************************
 
 // TODO: Rework this
-// RepeatAttribute should either
+// RetryAttribute should either
 //  1) Apply at load time to create the exact number of tests, or
 //  2) Apply at run time, generating tests or results dynamically
```
Comment (mikkelbu) [inline src/NUnitFramework/framework/Attributes/RetryAttribute.cs]: I think that we should delete all the comments regarding the TODO. As far as I can tell the comment is about `RepeatAttribute` and not `RetryAttribute`, e.g. we cannot at load time calculate the exact number of tests.

File: src/NUnitFramework/framework/Attributes/RetryAttribute.cs [hunk 2]
```diff
@@ -36,16 +36,16 @@
...
     [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
     public class RetryAttribute : PropertyAttribute, IWrapSetUpTearDown
     {
         private int _count;
 
         /// <summary>
-        /// Construct a RepeatAttribute
+        /// Construct a RetryAttribute
         /// </summary>
         /// <param name="count">The number of times to run the test</param>
```
Comment (mikkelbu) [inline src/NUnitFramework/framework/Attributes/RetryAttribute.cs]: Perhaps we should change the description of `count` to something like "The maximum number of times the test should be rerun if it fails".

Comment (mikkelbu) [inline src/NUnitFramework/framework/Attributes/RetryAttribute.cs]: We could also remove the same TODO in nunit/src/NUnitFramework/tests/Attributes/RetryAttributeTests.cs

Ps. You should remember to reference the issue number in the commitmessage or PR, like fixes #2313 such that the issue is automatically closed.

**Comment by mikkelbu** [comment]
@nunit/framework-team can I also kick the buildserver (aka Travis) or does that require special rights?

**Comment by CharliePoole** [comment]
It does. Kicked!

File: src/NUnitFramework/framework/Attributes/RetryAttribute.cs (hunk 2 above)
Comment (jnm2) [inline src/NUnitFramework/framework/Attributes/RetryAttribute.cs]: Also we should definitely start using `<see cref="RetryAttribute" />` here.

File: src/NUnitFramework/framework/Attributes/RetryAttribute.cs [hunk 3]
```diff
@@ -36,16 +36,16 @@
 namespace NUnit.Framework
 {
     /// <summary>
-    /// RepeatAttribute may be applied to test case in order
-    /// to run it multiple times.
+    /// RetryAttribute is used on a test method to specify that it should
```
Comment (jnm2) [inline src/NUnitFramework/framework/Attributes/RetryAttribute.cs]: We should definitely start using <see cref="RetryAttribute" /> here.

**Comment by ChrisMaddock** [comment]
@mikkelbu

I thought being a member of the framework team gave you access to restart builds? You'll need to log in to travis with your GitHub credentials. Once you've done that, there should be a 'restart build' button on the build page? This is what it looks like on mobile.

File: src/NUnitFramework/framework/Attributes/RetryAttribute.cs [hunk 4]
```diff
@@ -36,18 +29,18 @@
...
     public class RetryAttribute : PropertyAttribute, IWrapSetUpTearDown
     {
         private int _count;
 
         /// <summary>
-        /// Construct a RepeatAttribute
+        /// Construct a <see cref="RetryAttribute" />
         /// </summary>
-        /// <param name="count">The number of times to run the test</param>
+        /// <param name="count">The maximum number of times the test should be rerun if it fails</param>
```
Comment (jnm2) [inline src/NUnitFramework/framework/Attributes/RetryAttribute.cs]: Oh. This should say <code>The maximum number of times the test should be **run**</code>, not `rerun`.

**Comment by jnm2** [comment]
Well, we have an important preexisting problem to fix. I don't know if you want to do it in this PR or if we'll need a PR right after:

https://github.com/MhdTlb/nunit/blob/1446e65dc054dbc0711422b82baf70cace3f6f17/src/NUnitFramework/framework/Attributes/RetryAttribute.cs#L70-L80

Parameter `retryCount` and field `_retryCount` need to be renamed `tryCount` and `_tryCount`, and the param XML for the parameter needs to match the param XML for `RetryAttribute`'s parameter.

Also, it probably behooves us to rename `RetryAttribute`'s `count` parameter to `tryCount` just for clarity.

**Comment by MhdTlb** [comment]
@jnm2  I will try to fix it within this PR

**Pull request merged**